variables.tf

```diff
@@ -106,6 +106,11 @@ variable "network_policy" {
   default     = false
 }
 
+variable "network_policy_provider_enabled" {
```

Comment:
Does it ever make sense to have `network_policy=true` and `network_policy_provider_enabled=false`? If not I suggest we just use the single variable.